### ISSUE TITLE
Nav tree does not forget deleted pages

### ISSUE THREAD

**Issue by michaek** [michaek/middleman-navigation#1]
If you add a page while the preview server is running, then delete it, it's not removed from the rendered navigation.

**Comment by encorex** [comment]
Restarting the middleman server is the obvious workaround and is also needed when adding a new page. So I find this not to be too serious, although it would be nice if the middleman-navigation gem could to do this on the fly.

Maybe it would be interesting to see how others (other gems) implement this sort of functionality.

**Comment by michaek** [comment]
Yeah, totally. The thing is: I want this to work transparently for designers and content creators, so asking them to restart the server from the command line isn't ideal.
